File: LOST-FOUND/user-service/src/main/java/com/lostfound/user_service/service/UserService.java
package com.lostfound.user_service.service;

import com.lostfound.user_service.dto.LoginRequest;
import com.lostfound.user_service.dto.RegisterRequest;
import com.lostfound.user_service.dto.Response;

public interface UserService {

    Response register(RegisterRequest registerRequest);

    Response login(LoginRequest request);

    Response getProfile(String email);

    Response getAllUsers();

    Response getUserById(Long id);

    Response updateUser(Long id, RegisterRequest request);

    Response deleteUser(Long id);

    Response updateProfile(String email, RegisterRequest request);
}

File: frontend/src/services/userService.js
import api from './api'

export const getProfile = async () => {
  const response = await api.get('/user/profile')
  return response.data
}

export const updateProfile = async (userData) => {
  const response = await api.put('/user/profile', userData)
  return response.data
}

export const getAllUsers = async () => {
  const response = await api.get('/user/admin/all')
  return response.data
}

export const getUserById = async (id) => {
  const response = await api.get(`/user/admin/${id}`)
  return response.data
}

export const updateUser = async (id, userData) => {
  const response = await api.put(`/user/admin/${id}`, userData)
  return response.data
}

export const deleteUser = async (id) => {
  const response = await api.delete(`/user/admin/${id}`)
  return response.data
}
